searx/utils.py

```diff
@@ -28,37 +33,49 @@
 ecma_unescape4_re = re.compile(r'%u([0-9a-fA-F]{4})', re.UNICODE)
 ecma_unescape2_re = re.compile(r'%([0-9a-fA-F]{2})', re.UNICODE)
 
-xpath_cache = dict()
-lang_to_lc_cache = dict()
+xpath_cache = {}
+lang_to_lc_cache = {}
 
 
-class NotSetClass:
-    pass
+class NotSetClass:  # pylint: disable=too-few-public-methods
+    """Internal class for this module, do not create instance of this class.
+    Replace the None value, allow explicitly pass None as a function argument"""
 
 
 NOTSET = NotSetClass()
 
+STORAGE_UNIT_VALUE = {
```

Comment:
Two things I think we should start doing:

* Naming stuff that is not intended to be accessed from outside the module with a leading `_`.
* Putting global variables that are only used by one method directly above the method declaration.

Reply:
> Naming stuff that is not intended to be accessed from outside the module with a leading _.

Done.

> Putting global variables that are only used by one method directly above the method declaration (I think in general putting variable declarations as close as possible to their first use makes code more readable).

As soon as two or more function use a global variable, it becomes difficult to know where the variable is declared.

Reply:
> Naming stuff that is not intended to be accessed from outside the module with a leading _.

I don't think so / I don't see a difference between function names and names of other objects .. names that can accessed from outside are listed in `__all__` // [Setting `__all__` to an empty list indicates that the module has no public API.](https://www.python.org/dev/peps/pep-0008/#public-and-internal-interfaces)

Reply:
When looking at a function I want to be able to immediately tell whether it is private or public. I don't want to have to scroll up to look for a `__all__`.

Reply:
I can only repeat what is common practice, I don't know what a single person prefer.

Reply:
BTW: it is also a common practice to say a (function-) name that is not documented (by a docstring) is not public.

Reply:
Prefixing underscores is definitely common practice. It's also covered by PEP 8:

> _single_leading_underscore: weak "internal use" indicator. E.g. from M import * does not import objects whose names start with an underscore.

But even more importantly there is a [Pyright check](https://github.com/microsoft/pyright/blob/main/docs/configuration.md) for it that we can enable:

>reportPrivateUsage [boolean or string, optional]: Generate or suppress diagnostics for incorrect usage of private or protected variables or functions. Protected class members begin with a single underscore (“_”) and can be accessed only by subclasses. Private class members begin with a double underscore but do not end in a double underscore and can be accessed only within the declaring class. Variables and functions declared outside of a class are considered private if their names start with either a single or double underscore, and they cannot be accessed outside of the declaring module. The default value for this setting is 'none'.